source/slang/slang-check-conversion.cpp

```diff
@@ -613,19 +613,13 @@ bool SemanticsVisitor::_readAggregateValueFromInitializerList(
             }
             else
             {
-                // We don't know the element count statically,
-                // so what are we supposed to be doing?
-                //
-                if (outToExpr)
-                {
-                    getSink()->diagnose(
-                        fromInitializerListExpr,
-                        Diagnostics::cannotUseInitializerListForArrayOfUnknownSize,
-                        toElementCount);
-                }
-                return false;
-            }
+                auto toMakeArrayFromElementExpr = m_astBuilder->create<MakeArrayFromElementExpr>();
```

Comment:
This expr should store a valueExpr that represents the default value of Element style. If we cannot for. That valueExpr, then we need to issue an error.

Reply:
@csyonghe I'm planning on addressing this in the below fashion:

- If we cannot get a default value for a particular Element style, my understanding is that getDefaultVal() will return a LoweredValInfo with LoweredValInfo::val set to nullptr
- In the visitor implementation of MakeArrayFromElementExpr, we add a check for the above nullptr and if such a check gets triggered, we raise a SLANG_UNEXPECTED error stating that the default element for the element cannot be constructed.

Does this make sense?

Reply:
First of all, you can’t call anything that returns LoweredValInfo from the semantic visitor, if. Function returns LoweredValInfo, it belongs to IR lowering, not semantic checking. You will need to find something else that checks if a type is default constructable. Check with @kaizhangNV on what you need to call.

second,  we should never do SLANG_UNEXPECTED on anything that the user can write. It should be a proper diagnostic output.